### PR TITLE
Update Laravel application detection

### LINKED ISSUE #501: [2.x] Laravel is not detected when using Domain Oriented architecture
# How it started

I was running `php artisan test --parallel --recreate-databases` and it appeared that the databases were not recreated.

# What I found

`Illuminate\Testing\ParallelTesting::callSetUpProcessCallbacks` was never called, causing the databases not to be dropped (via the callback defined in `Illuminate\Testing\Concers\TestDatabases::bootTestDatabase`)

This was caused by the fact that the runner used by paratest was `Pest\Parallel\Paratest\Runner` instead of `Illuminate\Testing\ParallelRunner`.

The parallel plugin does not register the correct runner in `Pest\Parallel\Arguments\Laravel::editArguments` because `Pest\Parallel\Support\Environment::isALaravelApplication` does not detect the fact that it is a Laravel application.

# Why 

```php
final class Environment
{
    public static function isALaravelApplication(): bool
    {
        return class_exists('\App\Http\Kernel') && class_exists(ParallelRunner::class);
    }
}
```
The environment detects it is a Laravel application based on the fact that the `App\Http\Kernel` class exists. Unfortunately, using a Domain Oriented approach in Laravel, the `App\Http\Kernel` could be missing and the environment does not detect Laravel.
Renaming the `App` namespace could also make this test to fail.

# How to fix

I think it would be possible to detect the fact that it is a Laravel application using another test but I'm not sure what is the best one in order not to create other issues.

### PR DESCRIPTION
**Pull request by bastien-phi**
Fixes https://github.com/pestphp/pest/issues/501

This should work, tested locally in Laravel projects and some packages.

If someone has a better option, I can modify this PR.

Another way to test package install is to use `composer-runtime-api` and `\Composer\InstalledPackages`

### CODE REVIEW

File: src/Support/Environment.php [hunk 1]
```diff
@@ -13,6 +14,6 @@ final class Environment
 {
     public static function isALaravelApplication(): bool
     {
-        return class_exists('\App\Http\Kernel') && class_exists(ParallelRunner::class);
+        return class_exists(Application::class) && !class_exists(TestCase::class);
```
Comment (nunomaduro) [inline src/Support/Environment.php:17]: How about "class_exists(ParallelRunner::class)" too?

**Review by nunomaduro — CHANGES_REQUESTED** [review]

File: src/Support/Environment.php (hunk 1 above)
Comment (bastien-phi) [reply src/Support/Environment.php:17]: In the `composer.json` we have
```json
"conflict": {
    "laravel/framework": "<8.55"
},
``` 
and Laravel 8.55 and above has `ParallelRunner` so verifying that `laravel/framework` is installed should be enough.

If you want, I can modify this line to 
```php
return class_exists(Application::class) && class_exists(ParallelRunner::class) && !class_exists(TestCase::class);
```

Comment (bastien-phi) [reply src/Support/Environment.php:17]: @nunomaduro What do you think about that ?

**Comment by lukeraymonddowning** [comment]
Question: if a package developer (incorrectly) registers their `Package\\Tests` namespace under `autoload` instead of `autoload-dev` in their `composer.json` file, then won't Orchestra's `TestCase` be present in the autoloader? And if that is the case, won't this cause the same issue for a different set of circumstances?

**Comment by bastien-phi** [comment]
@lukeraymonddowning I don't think so. Orchestra's `TestCase` won't be present in autoloader because `orchertra/testbench` won't be installed.

However, this new implementation could fail if a package defines `orchestra/testbench` in its `require` instead of in `require-dev`. This could happen but does not makes sense. I don't know if `pest` should consider these edge cases caused by third parties.

**Comment by bastien-phi** [comment]
Hi @nunomaduro 

I just saw you closed this PR, maybe because you are working on V2.

In the meantime, the issue still exists. Are you opened to a new PR targetting V2 ?

**Comment by nunomaduro** [comment]
Yes. Yet, Pest v2.x is not yet ready for Parallel Testing. You can still do the PR if you want. cc @lukeraymonddowning

**Pull request closed without merging**